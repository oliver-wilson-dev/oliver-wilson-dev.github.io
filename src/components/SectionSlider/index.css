.tileContainer {
    max-height: none;
    display: flex;
    overflow: hidden;
    justify-content: flex-start;
}

.tileOnScreen {
    transform: translateX(0%);
    transition: all 0.2s ease;
}

.setWidth {
    box-sizing: border-box;
    width: 100%;
}

.offScreenTileRight {
    position: absolute;
    visibility: hidden;
    transform: translateX(110%);
    transition: all 0.2s ease;
}

.offScreenTileLeft {
    position: absolute;
    visibility: hidden;
    transform: translateX(-110%);
    transition: all 0.2s ease;
}

.arrowBtn {
    background-repeat: no-repeat;
    background-position: center;
    font-size: 50px;
    border: none;
    padding: 0;
    width: 50px;
    height: 50px;
    fill: var(--primaryBorderColour);
    transition-property: transform;
    transition-timing-function: ease;
    transition-duration: 0.5s;
    margin-right: 25px;

    &:last-child {
        margin-right: 0px;
    }

    &:not(.btnDisabled){
        &:hover {
            transform: scale(1.2);
        }
    }
}

.buttonWrapper {
    display: flex;
    justify-content: center;
    margin: 20px 0 30px 0;

    & .btnDisabled {
        opacity: 0.5;
    }
}
