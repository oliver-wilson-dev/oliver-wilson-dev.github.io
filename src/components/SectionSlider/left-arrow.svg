<?xml version="1.0" encoding="UTF-8"?>
<svg viewBox="0 0 12 22" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <g id="icons" stroke="none" stroke-width="1">
        <g id="Artboard" transform="translate(-527.000000, -101.000000)">
            <polygon id="left" transform="translate(538.250000, 112.250000) rotate(135.000000) translate(-538.250000, -112.250000) " points="530.928932 118.571068 530.928932 119.571068 545.571068 119.571068 545.571068 104.928932 544.571068 104.928932 544.571068 118.571068"></polygon>
        </g>
    </g>
</svg>