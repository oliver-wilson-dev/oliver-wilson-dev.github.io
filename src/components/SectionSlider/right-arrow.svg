<svg viewBox="0 0 12 22" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <defs></defs>
    <g id="icons" stroke="none" stroke-width="1">
        <g id="Artboard" transform="translate(-582.000000, -101.000000)">
            <polygon id="right" transform="translate(583.250000, 112.250000) rotate(-45.000000) translate(-583.250000, -112.250000) " points="575.928932 118.571068 575.928932 119.571068 590.571068 119.571068 590.571068 104.928932 589.571068 104.928932 589.571068 118.571068"></polygon>
        </g>
    </g>
</svg>