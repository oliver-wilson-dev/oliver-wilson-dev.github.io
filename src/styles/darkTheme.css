@import "./index.css";

[data-theme="DARK_THEME"] {
    --primaryBackground: #333;
    --primaryBackgroundImage: linear-gradient(#333, #333);
    --primaryBorderColour: #f8f8f8;
    --primaryTextColor: #f8f8f8;
    --spinnerTrailColour: rgba(248,248,248, 0.2);
    --sectionFadeColourFrom: rgba(51, 51, 51, 0);
    --sectionFadeColourTo: rgba(51, 51, 51, 1);
    --unvisitedLink: var(--projectGreen)
}