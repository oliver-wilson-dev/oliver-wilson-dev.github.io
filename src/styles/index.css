:root {
    --primaryBackgroundLight: #f8f8f8;
    --primaryBackgroundDark: #333;
    --primaryBackground: var(--primaryBackgroundLight);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: #333;

    --primaryTextColorDark: #f8f8f8;
    --primaryTextColorLight: #333;

    --primaryTextColorInverse: var(--primaryTextColorLight);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --linkBlue: rgb(21, 64, 204);
    --unvisitedLink: var(--linkBlue);

    --profilePictureBorder: var(--primaryBorderColour);

    --headerHeight: 65px;
    --headerPadding: 15px;

    --overlayBackground: rgba(0,0,0,.288);

    --projectGreen: #06d6a0;
    --projectPink: #EF476F;
    --projectYellow: #FFD166;
    --projectBlue: #118AB2;
    --projectDarkBlue: #073B4C;
}