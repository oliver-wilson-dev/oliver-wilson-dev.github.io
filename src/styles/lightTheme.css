@import "./index.css";

[data-theme="LIGHT_THEME"] {
    --primaryBackground: var(--primaryBackgroundLight);
    --primaryBackgroundInverse: var(--primaryBackgroundDark);

    --primaryBorderColour: #0b0a0a;
    --primaryBorderColourLight: #0b0a0a;

    --primaryTextColor: var(--primaryTextColorLight);
    --primaryTextColorInverse: var(--primaryTextColorDark);

    --spinnerTrailColour: rgba(11,10,10, 0.2);
    --sectionFadeColourFrom: rgba(248, 248, 248, 0);
    --sectionFadeColourTo: rgba(248, 248, 248, 1);

    --primaryBackgroundLight: #f8f8f8;

    --unvisitedLink: var(--projectGreen);
    --unvisitedLinkInverse: var(--projectGreen);
}