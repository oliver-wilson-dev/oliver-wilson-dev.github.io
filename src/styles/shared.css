.bodyFixed {
    position: fixed;
    left: 0;
    width: 100%;
    overflow-y: scroll;
}

.pageHeader {
    font-size: 5em;
    margin: 0;
    margin-top: 10px;
    background-clip: text;
    -webkit-background-clip: text;
    letter-spacing: 2px;
    text-align: center;
    width: 100%;
  }

  @media (min-width: 786px) {
    .landing--header {
    max-width: 800px;
    }
  }

  @media (min-width: 0px) and (max-width: 767px) {
    .pageHeader {
      font-size: 3em;
    }
  }