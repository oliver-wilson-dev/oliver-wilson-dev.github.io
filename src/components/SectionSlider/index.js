/* eslint-disable react/no-array-index-key */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import SectionTile from '../SectionTile';
import styles from './index.css';
import sectionStyles from '../SectionTile/index.css';
import LeftArrow from './left-arrow.svg';
import RightArrow from './right-arrow.svg';


const SectionSlider = ({ children }) => {
  const [currentSelectedIndex, setCurrentSelectedIndex] = useState(0);

  const firstElementSelected = currentSelectedIndex === 0;

  const lastElementSelected = currentSelectedIndex === children.length - 1;

  const increment = () => {
    if (!lastElementSelected) {
      setCurrentSelectedIndex(currentSelectedIndex + 1);
    }
  };
  const decrement = () => {
    if (!firstElementSelected) {
      setCurrentSelectedIndex(currentSelectedIndex - 1);
    }
  };

  return (
    <React.Fragment>
      <div className={styles.buttonWrapper}>

        <LeftArrow
          className={cn(styles.arrowBtn, {
            [styles.btnDisabled]: firstElementSelected
          })}
          onClick={decrement}
        />


        <RightArrow
          className={cn(styles.arrowBtn, {
            [styles.btnDisabled]: lastElementSelected
          })}
          onClick={increment}
        />
      </div>
      <div className={cn(styles.tileContainer, sectionStyles.displayBox)}>
        { children.map((child, index) => {
          const props = {
            additionalStyles: {
              [styles.removeBorder]: true,
              [styles.tileOnScreen]: index === currentSelectedIndex,
              [styles.offScreenTileRight]: index > currentSelectedIndex,
              [styles.offScreenTileLeft]: index < currentSelectedIndex
            }
          };

          return (
            <SectionTile key={index} {...props}>
              {child}
            </SectionTile>
          );
        })}
      </div>
    </React.Fragment>
  );
};

SectionSlider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default SectionSlider;
