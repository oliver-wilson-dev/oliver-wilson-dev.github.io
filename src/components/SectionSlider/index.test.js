import React from 'react';
import { shallow } from 'enzyme';
import SectionSlider from '.';
import styles from './index.css';

jest.mock('./left-arrow.svg', () => {
  const LeftArrow = () => null;

  return LeftArrow;
});

jest.mock('./right-arrow.svg', () => {
  const RightArrow = () => null;

  return RightArrow;
});

const defaultChildren = [
  <p key="1">hello world</p>,
  <p key="2">goodbye world</p>
];

const render = (children = defaultChildren) => shallow(
  <SectionSlider>
    {children}
  </SectionSlider>
);

describe('SectionSlider', () => {
  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when clicking the right arrow', () => {
    it('should apply the btnDisabled class to the right arrow', () => {
      const component = render();

      component.find(`.${styles.arrowBtn}`).at(1).simulate('click');

      component.update();

      expect(component.find(`.${styles.arrowBtn}`).at(1).hasClass(styles.btnDisabled)).toBe(true);
    });

    describe('and then clicking it again', () => {
      it('should render correctly', () => {
        const component = render();

        component.find(`.${styles.arrowBtn}`).at(1).simulate('click');

        component.update();

        component.find(`.${styles.arrowBtn}`).at(1).simulate('click');

        component.update();

        expect(component).toMatchSnapshot();
      });
    });

    describe('and then clicking the left arrow', () => {
      it('should apply the btnDisabled class to the left arrow', () => {
        const component = render();

        component.find(`.${styles.arrowBtn}`).at(1).simulate('click');

        component.update();

        component.find(`.${styles.arrowBtn}`).at(0).simulate('click');

        component.update();

        expect(component.find(`.${styles.arrowBtn}`).at(0).hasClass(styles.btnDisabled)).toBe(true);
      });

      describe('and then clicking it again', () => {
        it('should render correctly', () => {
          const component = render();

          component.find(`.${styles.arrowBtn}`).at(1).simulate('click');

          component.update();

          component.find(`.${styles.arrowBtn}`).at(0).simulate('click');

          component.update();

          component.find(`.${styles.arrowBtn}`).at(0).simulate('click');

          component.update();

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
