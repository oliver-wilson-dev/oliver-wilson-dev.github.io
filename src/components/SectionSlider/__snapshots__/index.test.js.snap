// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionSlider should render correctly 1`] = `
<Fragment>
  <div
    className="buttonWrapper"
  >
    <LeftArrow
      className="arrowBtn btnDisabled"
      onClick={[Function]}
    />
    <RightArrow
      className="arrowBtn"
      onClick={[Function]}
    />
  </div>
  <div
    className="tileContainer displayBox"
  >
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": false,
          "offScreenTileRight": false,
          "removeBorder": true,
          "tileOnScreen": true,
        }
      }
      key="0"
    >
      <p
        key="1"
      >
        hello world
      </p>
    </SectionTile>
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": false,
          "offScreenTileRight": true,
          "removeBorder": true,
          "tileOnScreen": false,
        }
      }
      key="1"
    >
      <p
        key="2"
      >
        goodbye world
      </p>
    </SectionTile>
  </div>
</Fragment>
`;

exports[`SectionSlider when clicking the right arrow and then clicking it again should render correctly 1`] = `
<Fragment>
  <div
    className="buttonWrapper"
  >
    <LeftArrow
      className="arrowBtn"
      onClick={[Function]}
    />
    <RightArrow
      className="arrowBtn btnDisabled"
      onClick={[Function]}
    />
  </div>
  <div
    className="tileContainer displayBox"
  >
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": true,
          "offScreenTileRight": false,
          "removeBorder": true,
          "tileOnScreen": false,
        }
      }
      key="0"
    >
      <p
        key="1"
      >
        hello world
      </p>
    </SectionTile>
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": false,
          "offScreenTileRight": false,
          "removeBorder": true,
          "tileOnScreen": true,
        }
      }
      key="1"
    >
      <p
        key="2"
      >
        goodbye world
      </p>
    </SectionTile>
  </div>
</Fragment>
`;

exports[`SectionSlider when clicking the right arrow and then clicking the left arrow and then clicking it again should render correctly 1`] = `
<Fragment>
  <div
    className="buttonWrapper"
  >
    <LeftArrow
      className="arrowBtn btnDisabled"
      onClick={[Function]}
    />
    <RightArrow
      className="arrowBtn"
      onClick={[Function]}
    />
  </div>
  <div
    className="tileContainer displayBox"
  >
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": false,
          "offScreenTileRight": false,
          "removeBorder": true,
          "tileOnScreen": true,
        }
      }
      key="0"
    >
      <p
        key="1"
      >
        hello world
      </p>
    </SectionTile>
    <SectionTile
      additionalStyles={
        Object {
          "offScreenTileLeft": false,
          "offScreenTileRight": true,
          "removeBorder": true,
          "tileOnScreen": false,
        }
      }
      key="1"
    >
      <p
        key="2"
      >
        goodbye world
      </p>
    </SectionTile>
  </div>
</Fragment>
`;
